### ISSUE TITLE
[bug] CMakeDeps generated .cmake config files doesn't put linker flags in quotation

### ISSUE THREAD

**Issue by fanselm** [conan-io/conan#9936]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 16.04
  * Compiler+version: GCC 9.3
  * Conan version: 1.40.1
  * Python version: 3.5.2

### Steps to reproduce (Include if Applicable)

Create a package that specifies linker flags, e.g: MyLib

```
def package_info(self):
    self.cpp_info.libs = ["mylib"]
    self.cpp_info.sharedlinkflags = ["-z now", "-z relro"]
    self.cpp_info.exelinkflags = ["-z now", "-z relro"]
```

Then for a package that depends on MyLib we rely on CMakeDeps generator

```
[requires]
mylib/1.0.0

[generators]
CMakeDeps
```

It then has a CMakeLists.txt that uses the library:

```
cmake_minimum_required(VERSION 3.15)
project(CMakeDepsTest C)

# Prefer -config.cmake files to be ConanDeps friendly
set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
list(APPEND CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}")

find_package(MYLIB)

add_executable(mytest main.c)
target_link_libraries(mytest mylib::mylib)
```

When building this test application the linker will fail with errors:
```
/usr/bin/ld: warning: -z --library=now ignored.
/usr/bin/ld: cannot find -lrelro
```

Inspecting the linker command line it's clear that the linker args got passed wrong, i.e. CMake thinks they are a list of libraries to be linked in. The reason for this is that the generated file `mylib-release-x86_64-data.cmake` specifies:
```
set(mylib_SHARED_LINK_FLAGS_RELEASE -z now;-z relro)
```
which is treated as a list "-z;now;-z;relro;". Now this further goes wrong because the linker flags are added as libraries in the file `mylib-Target-release.cmake`:
```
set_property(TARGET intel-mpi::intel-mpi
             PROPERTY INTERFACE_LINK_LIBRARIES
             $<$<CONFIG:Release>:${intel-mpi_LIBRARIES_TARGETS_RELEASE}
                                           ${intel-mpi_LINKER_FLAGS_RELEASE}> APPEND)
```
and not populating the `INTERFACE_LINK_OPTIONS` property.

The solution is clear:
1) Make sure `mylib_SHARED_LINK_FLAGS_RELEASE` is set as a list of strings or a single string
2) Set linker flags by populating the `INTERFACE_LINK_OPTIONS` property instead of appending them to the `INTERFACE_LINK_LIBRARIES`.

**Comment by memsharded** [comment]
This is true, the ``sharedlinkflags`` is not prepared to contain things with spaces, this has to be fixed.

Please @franramirez688 try the following:

- Lets build a full functional test, that uses ``exelinkflags``, but also ``cxxflags``
- Real test, building something. 
- Only linux
- The dependency defining cpp_info can be empty, only injecting flags
- Use several flags, some using spaces too
- Check the flags are correctly applied both compile and link

**Comment by memsharded** [comment]
Then, changing the cmake to use INTERFACE_LINK_OPTIONS seems easy, and it is compatible CMake 3.15:

```diff
--- a/conan/tools/cmake/cmakedeps/templates/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps/templates/target_configuration.py
@@ -110,8 +110,10 @@ class TargetConfigurationTemplate(CMakeDepsFileTemplate):
         set_property(TARGET {{target_namespace}}::{{global_target_name}}
                      PROPERTY INTERFACE_LINK_LIBRARIES
                      $<$<CONFIG:{{configuration}}>:${{'{'}}{{pkg_name}}_LIBRARIES_TARGETS{{config_suffix}}}
-                                                   ${{'{'}}{{pkg_name}}_LINKER_FLAGS{{config_suffix}}}
                                                    ${{'{'}}{{pkg_name}}_OBJECTS{{config_suffix}}}> APPEND)
+        set_property(TARGET {{target_namespace}}::{{global_target_name}}
+                     PROPERTY INTERFACE_LINK_OPTIONS
+                     $<$<CONFIG:{{configuration}}>:${{'{'}}{{pkg_name}}_LINKER_FLAGS{{config_suffix}}}> APPEND)
         set_property(TARGET {{target_namespace}}::{{global_target_name}}
                      PROPERTY INTERFACE_INCLUDE_DIRECTORIES
                      $<$<CONFIG:{{configuration}}>:${{'{'}}{{pkg_name}}_INCLUDE_DIRS{{config_suffix}}}> APPEND)
@@ -129,8 +131,9 @@ class TargetConfigurationTemplate(CMakeDepsFileTemplate):
         ########## COMPONENT {{ comp_name }} TARGET PROPERTIES ######################################
         set_property(TARGET {{ target_namespace }}::{{ comp_name }} PROPERTY INTERFACE_LINK_LIBRARIES
                      $<$<CONFIG:{{ configuration }}>:{{tvalue(pkg_name, comp_name, 'LINK_LIBS', config_suffix)}}
-                     {{tvalue(pkg_name, comp_name, 'LINKER_FLAGS', config_suffix)}}
                      {{tvalue(pkg_name, comp_name, 'OBJECTS', config_suffix)}}> APPEND)
+        set_property(TARGET {{ target_namespace }}::{{ comp_name }} PROPERTY INTERFACE_LINK_OPTIONS
+                     $<$<CONFIG:{{ configuration }}>:{{tvalue(pkg_name, comp_name, 'LINKER_FLAGS', config_suffix)}}> APPEND)
         set_property(TARGET {{ target_namespace }}::{{ comp_name }} PROPERTY INTERFACE_INCLUDE_DIRECTORIES
                      $<$<CONFIG:{{ configuration }}>:{{tvalue(pkg_name, comp_name, 'INCLUDE_DIRS', config_suffix)}}> APPEND)
         set_property(TARGET {{ target_namespace }}::{{ comp_name }} PROPERTY INTERFACE_COMPILE_DEFINITIONS
```

**Comment by fanselm** [comment]
Hi @memsharded , thanks for reacting so quickly. Let me know if I can help with anything.

**Comment by memsharded** [comment]
Merged https://github.com/conan-io/conan/pull/9980, will be released next 1.43
